### ISSUE TITLE
Reverted versions overriding updated version

### ISSUE THREAD

**Issue by estelle** [mdn/kuma#5976]
https://wiki.developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Styling_HTML_forms

I edited the page and saved it. 
I then continued to edit, and saved it again.
When I saved it, it saved a previous version of the page rather than the document i was saving.

This is the second time something like this has happened in this one document. Previous time was during the migration to the new version, and those changes were lost - there is no trace like there was this time.
But, since something similar happened again and this time it left a trace (in the history), i'm reporting.

**Comment by peterbe** [comment]
Note-to-self; https://wiki.developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Styling_HTML_forms$history
and it looks like this:
<img width="1140" alt="Screen Shot 2019-10-14 at 12 19 10 PM" src="https://user-images.githubusercontent.com/26739/66766743-da758100-ee7c-11e9-9dd1-0a7e0fd9161d.png">

So you submitted 2 saves (I.e. presses on "Publish and Keep Editing"). The first press updated the page and a row was added to the $history page. But the second press was entirely lost with no record of it anywhere?

**Comment by estelle** [comment]
This has happened twice. The first time it was during the switchover so I thought it had to do with that. But, since it happened again, i filed a bug. I am not 100% sure what happened, but my feeling is that it had to do with uploading an image. 

It seems that the two times the text that was saved when I uploaded an image was not the current version of the page. Not sure where it would have pulled an old version of the page though.

**Comment by peterbe** [comment]
I'm sorry for being dim but I'm still not sure I get what happened. Did you hit the save button ("Publish" or "Publish and Keep Editing") and it went into oblivion? Or did it save fine but not updating the right revision?

Looking at https://wiki.developer.mozilla.org/en-US/docs/Learn/HTML/Forms/Styling_HTML_forms$history there are 9 since (and including Oct 9). Does that reflect you pressing the Save button 9 times? Or did you press more than 9 times and it simply lost one of those?

Also, doesn't the editor have some sort of auto-save functionality? Was that able to save any lost edits?

Basically, I fear this is the nature of a WYSIWYG solution and it always has been and always will be. More reasons for why it'd be

**Comment by peterbe** [comment]
WYSIWYG forms are an abstraction of a web textarea form. If the server isn't working to receive it there is always a risk that the content of the textarea gets lost. 

Here, I opened the Wiki editor, typed a couple things and then went to my `docker-compose` terminal and killed the Kuma server. Then, then I press the "Publish and Keep Editing" button this happens:
<img width="1616" alt="Screen Shot 2019-10-17 at 1 49 45 PM" src="https://user-images.githubusercontent.com/26739/67034177-35111600-f0e5-11e9-9747-cf250d391ad2.png">

Did that happen to you?

If you, instead, open the wiki editor, type a couple of things and then kill `docker-compose` in the terminal, and then press the "Publish" button, this happens:
<img width="1600" alt="Screen Shot 2019-10-17 at 1 57 38 PM" src="https://user-images.githubusercontent.com/26739/67034731-38f16800-f0e6-11e9-9de0-5c38a0949233.png">

That HTTP POST submission hits a brick wall. The same could happen if Kuma stays up but your own internet connection or DNS has a sudden and complete failure. 
If you, from this point, press the Back button, the wiki form will, it appears, reload the page from the server and losing your changes. This is because it's not a regular textarea. 

Perhaps these are just loose thoughts that isn't news but it definitely seems that the "Publish" button is a LOT more dangerous and the "Publish and Keep Editing" since the latter at least doesn't loose the content of the WYSIWYG textarea.

**Comment by peterbe** [comment]
The other option is that the server rather poorly accepted the HTTP POST and returned a 200 OK but internally actually failed to properly save the edit in MySQL. 

I've seen many scary areas deep in Kuma where database transactions aren't used and I've seen various pieces of code where exceptions are dangerously swallowed. 

Without knowing more about exactly what happened so we can develop steps to reproduce, I think we should close this issue and focus on the future instead; Stumptown. :)

**Comment by tobinmori** [comment]
Hi @estelle are you still having issues with saving? Or do these seem to have been anomalies?

**Comment by estelle** [comment]
OK, so it happened again. This time I *DID* get the unable to save box, but reverted at the same time, so i lost what i was trying to save. Maybe this is what happened the other times.

**Comment by peterbe** [comment]
> but reverted at the same time

Oh my! Not only did the POST fail, it also changed the current revision of the document?!
Are your changes safe now though? Did you copy it after you got the error growl notification?
Is your submission listed in the revision history?

**Comment by atopal** [comment]
Hey @estelle do you still remember the incidence to answer Peter's questions? It would be helpful to triage this bug. Thank you!

**Comment by atopal** [comment]
Don't have enough info to triage.
